services/time/src/lib.rs

```diff
@@ -111,62 +113,114 @@ transactions! {
     }
 }
 
-impl Transaction for TxTime {
-    fn verify(&self) -> bool {
-        self.verify_signature(self.pub_key())
+#[derive(Debug)]
+#[repr(u8)]
+enum ErrorCode {
+    UnknownSender = 0,
+    ValidatorTimeIsLonger = 1,
+}
+
+#[derive(Debug, Fail)]
+enum Error {
+    #[fail(display = "Not authored by a validator")]
+    UnknownSender,
+
+    #[fail(display = "The validator time is longer than the proposed one")]
+    ValidatorTimeIsLonger,
+}
```

Comment:
Currently, `ErrorCode` and `Error` enums are exact duplicates. I understand that, in the future, we might want to extend `Error` with variants with data, but I would suggest using a single `enum` until that time.